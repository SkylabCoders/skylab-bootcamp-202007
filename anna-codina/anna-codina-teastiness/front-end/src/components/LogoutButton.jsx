import React from 'react';
import { useAuth0 } from '@auth0/auth0-react';
import { logoutUser } from '../actions/userActions';

const LogoutButton = () => {
  const { logout } = useAuth0();

  return (
    <button
      className="auth-button"
      onClick={() => {
        logoutUser();
        logout({ returnTo: window.location.origin });
      }}
    >
      Log Out
    </button>
  );
};

export default LogoutButton;
