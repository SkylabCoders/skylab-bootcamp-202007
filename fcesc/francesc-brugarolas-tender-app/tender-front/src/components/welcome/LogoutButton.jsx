import React from "react";
import { useAuth0 } from "@auth0/auth0-react";
import { removeCurrentUser } from './../../redux/actions/userActions';
import { store } from '../../index';

const LogoutButton = () => {
  const { logout } = useAuth0();

  return (
    <button className='btn btn-default btn-centered' onClick={() => {
        sessionStorage.removeItem('USER_TOKEN');
        store.dispatch(removeCurrentUser())
        logout({ returnTo: window.location.origin })
      }}>
      Log Out
    </button>
  );
};

export default LogoutButton;