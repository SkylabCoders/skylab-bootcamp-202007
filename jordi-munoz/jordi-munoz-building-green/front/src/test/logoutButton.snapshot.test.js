import renderer from 'react-test-renderer';
import React from 'react';
import LogoutButton from '../components/logoutButton';
import { BrowserRouter } from 'react-router-dom';

describe('logoutButton snapshot', () => {
  const logoutButtonTree = renderer.create(
    <BrowserRouter>
      <LogoutButton />
    </BrowserRouter>
  );
  it('should match', () => {
    expect(logoutButtonTree.toJSON()).toMatchSnapshot();
  });

});