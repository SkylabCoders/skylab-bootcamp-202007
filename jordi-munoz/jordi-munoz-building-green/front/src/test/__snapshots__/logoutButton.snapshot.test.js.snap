// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logoutButton snapshot should match 1`] = `
<button
  className="logoutButton"
  onClick={[Function]}
>
  Log Out
</button>
`;
